drivers/hud_data_quality_report/app/controllers/hud_data_quality_report/base_controller.rb

```diff
@@ -8,47 +8,32 @@ module HudDataQualityReport
   class BaseController < ::HudReports::BaseController
     before_action :filter
 
-    def filter_params
-      return {} unless params[:filter]
-
-      filter_p = params.require(:filter).permit(@filter.known_params)
-      filter_p[:user_id] = current_user.id
-
-      filter_p
+    def available_report_versions
+      {
+        'FY 2020' => :fy2020,
+        # 'FY 2021' => :fy2021,
+      }.freeze
     end
+    helper_method :available_report_versions
 
-    private def filter
-      year = if Date.current.month >= 10
-        Date.current.year
-      else
-        Date.current.year - 1
-      end
-      # Some sane defaults, using the previous report if available
-      @filter = filter_class.new(user_id: current_user.id)
-      if filter_params.blank?
-        prior_report = generator.find_report(current_user)
-        options = prior_report&.options
-        site_coc_codes = GrdaWarehouse::Config.get(:site_coc_codes).presence&.split(/,\s*/)
-        if options.present?
-          @filter.start = options['start'].presence || Date.new(year - 1, 10, 1)
-          @filter.end = options['end'].presence || Date.new(year, 9, 30)
-          @filter.coc_codes = options['coc_codes'].presence || site_coc_codes
-          @filter.update(options.with_indifferent_access)
-        else
-          @filter.start = Date.new(year - 1, 10, 1)
-          @filter.end = Date.new(year, 9, 30)
-        end
-      end
-      # Override with params if set
-      @filter.update(filter_params) if filter_params.present?
+    def default_report_version
+      :fy2020
     end
 
     private def filter_class
-      generator.filter_class
+      ::HudDataQualityReport::Filters::DqFilter
```

Comment:
I realize I don't understand this change -- does the generator no longer need to return these? Or is it sometime nil?

Reply:
I think the issues is that it's a catch 22.  `def filter` which sets the filter uses the `generator` and `def generator` calls `@filter` so there was some infinite recursion.  Seemed easiest to just hard code this class since it's the same for all controllers in the driver.